src/cookie.rs

```diff
@@ -0,0 +1,220 @@
+//! The cookies module contains types for working with request and response cookies.
+
+use cookie_crate;
+use header;
+use std::borrow::Cow;
+use std::fmt;
+use std::time::SystemTime;
+
+/// Convert a time::Tm time to SystemTime.
+fn tm_to_systemtime(tm: ::time::Tm) -> SystemTime {
+    let seconds = tm.to_timespec().sec;
+    let duration = std::time::Duration::from_secs(seconds.abs() as u64);
+    if seconds > 0 {
+        SystemTime::UNIX_EPOCH + duration
+    } else {
+        SystemTime::UNIX_EPOCH - duration
+    }
+}
+
+/// Convert a SystemTime to time::Tm.
+/// Returns None if the conversion failed.
+fn systemtime_to_tm(time: SystemTime) -> Option<::time::Tm> {
+    let seconds = match time.duration_since(SystemTime::UNIX_EPOCH) {
+        Ok(duration) => duration.as_secs() as i64,
+        Err(_) => {
+            if let Ok(duration) = SystemTime::UNIX_EPOCH.duration_since(time) {
+                (duration.as_secs() as i64) * -1
+            } else {
+                return None;
+            }
+        }
+    };
+    Some(::time::at_utc(::time::Timespec::new(seconds, 0)))
+}
+
+/// Represents the 'SameSite' attribute of a cookie.
+#[derive(PartialEq, Eq, Clone, Copy, Debug)]
+pub enum SameSite {
```

Comment:
Since this is a relatively new attribute for cookies, it might be safer to not expose it as an enum. If there is need for determine `same-site`, perhaps we could have specific accessor methods instead? `is_same_site_strict()`?

Reply:
I mostly just copied the setters from `cookie`.

They start making sense once there is a `CookieStore::add(mycookie)` method.
But we can leave them out for now.

Reply:
removed the setters and the SameSite wrapper